<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="generator" content="pandoc" />
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <meta name="author" content="Jenny Leopoldina Smith" />
    <meta name="date" content="2021-06-03" />
    <title>A Modified Poster</title>

        <script src="pagedown_tutorial_files/header-attrs-2.7/header-attrs.js"></script>
        <link href="pagedown_tutorial_files/paged-0.14.6/css/poster-relaxed.css" rel="stylesheet" />
        <script src="pagedown_tutorial_files/kePrint-0.0.1/kePrint.js"></script>
        <link href="pagedown_tutorial_files/lightable-0.0.1/lightable.css" rel="stylesheet" />
    
    
    <style type="text/css">pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */</style>

    
  </head>
  <body>


<div id="multi-omics-clinical-reporting-with-rmarkdown" class="section level1">
<h1>Multi-Omics Clinical Reporting <span class="tiny">with</span> Rmarkdown</h1>
<img src="%22assets/omicsReportR.png%22" title="fig:" alt="a flowchart" />
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-2"></span>
<img src="assets/omicsReportR.png" alt="snpReportR: A Method for RNAseq Variant Detection Reporting" width="100%" />
<p class="caption">
Figure 1: snpReportR: A Method for RNAseq Variant Detection Reporting
</p>
</div>
</div>
<div id="patient-information" class="section level1">
<h1><span class="red">Patient Information</span></h1>
<p>This section provides information on the patient and their relevant clinical information.</p>
<div class="member-cards">
<div id="patient-anonymous-id" class="section level2">
<h2>Patient Anonymous ID</h2>
<p>this is a subtitle.</p>
<p>Need to provide a table here.</p>
</div>
<div id="patient-dob" class="section level2">
<h2>Patient DOB</h2>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; ">
<caption>
<span id="tab:unnamed-chunk-3">Table 1: </span>Patient details:
</caption>
<thead>
<tr>
<th style="text-align:left;">
Information
</th>
<th style="text-align:left;">
Value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Name:
</td>
<td style="text-align:left;">
Jane Doe
</td>
</tr>
<tr>
<td style="text-align:left;">
DOB:
</td>
<td style="text-align:left;">
01/01/1900
</td>
</tr>
<tr>
<td style="text-align:left;">
Sex:
</td>
<td style="text-align:left;">
F
</td>
</tr>
<tr>
<td style="text-align:left;">
Genetic Tests:
</td>
<td style="text-align:left;">
RNA, DNA, and TCR sequencing
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="disclaimer">All data is protected under HIPAA and . This report (poster but trying to use the css and html for this to learn how to make my own templates) was created via the R package <a href="https://github.com/rstudio/pagedown"><strong>pagedown</strong></a>.</span></p>
</div>
<div id="genetic-tests" class="section level1">
<h1><span class="blue">Genetic Tests</span></h1>
<div id="genetic-testing-performed" class="section level2">
<h2>Genetic testing performed</h2>
<p>Genetic testing was carried out using RNA-sequencing, DNA-sequencing, and TCR sequencing. These data were used to determine the expressed variants for single nucleotide variants (SNVs) and single nucleotide polymorphisms (SNPs) by transcriptomics, and also allowed one to determine both somatic and germline variants using DNA. In addition, large chromosomal rearrangment, called structural variants (SVs), were identifed using [DNA or RNA?] sequencing. Finally, TCR sequencing allows one to investigate how immune system/inflammation maybe affected due to disease or other underlying conditions.</p>
<div class="figure">
<img src="https://user-images.githubusercontent.com/41301333/120476740-dd4f1d80-c3a2-11eb-9c81-248f56bfb84f.png" alt="" />
<p class="caption">The genetic testing results are compiled and the report is produced with <strong>OmicsReportR</strong></p>
</div>
</div>
<div id="test-information" class="section level2">
<h2>Test Information</h2>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; ">
<caption>
<span id="tab:unnamed-chunk-5">Table 2: </span>Test details:
</caption>
<thead>
<tr>
<th style="text-align:left;">
Testing.Methodolgy
</th>
<th style="text-align:left;">
Description
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
DNA-seq:
</td>
<td style="text-align:left;">
DNA
</td>
</tr>
<tr>
<td style="text-align:left;">
RNA-seq:
</td>
<td style="text-align:left;">
RNA
</td>
</tr>
<tr>
<td style="text-align:left;">
TCR-seq:
</td>
<td style="text-align:left;">
T-Cell Receptor sequencing
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="schedule" class="section level1">
<h1><span class="pink">Schedule</span></h1>
<div id="outline-2018-12-15" class="section level2">
<h2>Outline (2018-12-15)</h2>
<p>Genetic testing was carried out using RNA-sequencing, DNA-sequencing, and TCR sequencing. These data were used to determine the expressed variants for single nucleotide variants (SNVs) and single nucleotide polymorphisms (SNPs) by transcriptomics, and also allowed one to determine both somatic and germline variants using DNA. In addition, large chromosomal rearrangment, called structural variants (SVs), were identifed using [DNA or RNA?] sequencing. Finally, TCR sequencing allows one to investigate how immune system/inflammation maybe affected due to disease or other underlying conditions.</p>
</div>
<div id="content-2019-03-01" class="section level2">
<h2>Content (2019-03-01)</h2>
<p>Genetic testing was carried out using RNA-sequencing, DNA-sequencing, and TCR sequencing. These data were used to determine the expressed variants for single nucleotide variants (SNVs) and single nucleotide polymorphisms (SNPs) by transcriptomics, and also allowed one to determine both somatic and germline variants using DNA. In addition, large chromosomal rearrangment, called structural variants (SVs), were identifed using [DNA or RNA?] sequencing. Finally, TCR sequencing allows one to investigate how immune system/inflammation maybe affected due to disease or other underlying conditions.</p>
</div>
<div id="review-and-revision-2019-03-31" class="section level2">
<h2>Review and revision (2019-03-31)</h2>
<p>Genetic testing was carried out using RNA-sequencing, DNA-sequencing, and TCR sequencing. These data were used to determine the expressed variants for single nucleotide variants (SNVs) and single nucleotide polymorphisms (SNPs) by transcriptomics, and also allowed one to determine both somatic and germline variants using DNA. In addition, large chromosomal rearrangment, called structural variants (SVs), were identifed using [DNA or RNA?] sequencing. Finally, TCR sequencing allows one to investigate how immune system/inflammation maybe affected due to disease or other underlying conditions.</p>
</div>
<div id="copyediting-2019-04-20" class="section level2">
<h2>Copyediting (2019-04-20)</h2>
<p>Genetic testing was carried out using RNA-sequencing, DNA-sequencing, and TCR sequencing. These data were used to determine the expressed variants for single nucleotide variants (SNVs) and single nucleotide polymorphisms (SNPs) by transcriptomics, and also allowed one to determine both somatic and germline variants using DNA. In addition, large chromosomal rearrangment, called structural variants (SVs), were identifed using [DNA or RNA?] sequencing. Finally, TCR sequencing allows one to investigate how immune system/inflammation maybe affected due to disease or other underlying conditions.</p>
</div>
<div id="publishing-2019-05-30" class="section level2">
<h2>Publishing (2019-05-30)</h2>
<p>Genetic testing was carried out using RNA-sequencing, DNA-sequencing, and TCR sequencing. These data were used to determine the expressed variants for single nucleotide variants (SNVs) and single nucleotide polymorphisms (SNPs) by transcriptomics, and also allowed one to determine both somatic and germline variants using DNA. In addition, large chromosomal rearrangment, called structural variants (SVs), were identifed using [DNA or RNA?] sequencing. Finally, TCR sequencing allows one to investigate how immune system/inflammation maybe affected due to disease or other underlying conditions.</p>
</div>
</div>
<div id="contents" class="section level1">
<h1><span class="green">Contents</span></h1>
<p>Genetic testing was carried out using RNA-sequencing, DNA-sequencing, and TCR sequencing. These data were used to determine the expressed variants for single nucleotide variants (SNVs) and single nucleotide polymorphisms (SNPs) by transcriptomics, and also allowed one to determine both somatic and germline variants using DNA. In addition, large chromosomal rearrangment, called structural variants (SVs), were identifed using [DNA or RNA?] sequencing. Finally, TCR sequencing allows one to investigate how immune system/inflammation maybe affected due to disease or other underlying conditions.</p>
<p>Genetic testing was carried out using RNA-sequencing, DNA-sequencing, and TCR sequencing. These data were used to determine the expressed variants for single nucleotide variants (SNVs) and single nucleotide polymorphisms (SNPs) by transcriptomics, and also allowed one to determine both somatic and germline variants using DNA. In addition, large chromosomal rearrangment, called structural variants (SVs), were identifed using [DNA or RNA?] sequencing. Finally, TCR sequencing allows one to investigate how immune system/inflammation maybe affected due to disease or other underlying conditions.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># some nice R code here</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">lm</span>(dist <span class="sc">~</span> speed, cars)</span></code></pre></div>
<p>Genetic testing was carried out using RNA-sequencing, DNA-sequencing, and TCR sequencing. These data were used to determine the expressed variants for single nucleotide variants (SNVs) and single nucleotide polymorphisms (SNPs) by transcriptomics, and also allowed one to determine both somatic and germline variants using DNA. In addition, large chromosomal rearrangment, called structural variants (SVs), were identifed using [DNA or RNA?] sequencing. Finally, TCR sequencing allows one to investigate how immune system/inflammation maybe affected due to disease or other underlying conditions.</p>
<div class="figure">
<img src="https://github.com/yihui/xaringan/releases/download/v0.0.2/karl-moustache.jpg" alt="" />
<p class="caption">The most well-known feature of <strong>xaringan</strong>: the random Moustache Karl (aka <code>yolo = TRUE</code>).</p>
</div>
<p>Genetic testing was carried out using RNA-sequencing, DNA-sequencing, and TCR sequencing. These data were used to determine the expressed variants for single nucleotide variants (SNVs) and single nucleotide polymorphisms (SNPs) by transcriptomics, and also allowed one to determine both somatic and germline variants using DNA. In addition, large chromosomal rearrangment, called structural variants (SVs), were identifed using [DNA or RNA?] sequencing. Finally, TCR sequencing allows one to investigate how immune system/inflammation maybe affected due to disease or other underlying conditions.</p>
<p><span class="math display">\[\bar{X}=\frac{1}{n}\sum_{i=1}^nX_i\]</span></p>
<p>Genetic testing was carried out using RNA-sequencing, DNA-sequencing, and TCR sequencing. These data were used to determine the expressed variants for single nucleotide variants (SNVs) and single nucleotide polymorphisms (SNPs) by transcriptomics, and also allowed one to determine both somatic and germline variants using DNA. In addition, large chromosomal rearrangment, called structural variants (SVs), were identifed using [DNA or RNA?] sequencing. Finally, TCR sequencing allows one to investigate how immune system/inflammation maybe affected due to disease or other underlying conditions.</p>
</div>
<div id="references" class="section level1">
<h1><span class="yellow">References</span></h1>
</div>

<script type="text/javascript">
(function() {
  // add appropriate classes to h1 and its parent div
  var i, j, d, s, el, els = document.getElementsByTagName('h1');
  for (i = 0; i < els.length; i++) {
    el = els[i];
    el.parentNode.className += ' section-' + (i + 1);
    if (i === 0) continue;  // that's all we need for the poster title
    el.className += ' ui title ribbon label';  // poster sections
    s = el.firstElementChild;  // move the class from inner <span> to <h1>
    if (s) {
      el.className += ' ' + s.className;
      s.outerHTML = s.innerHTML;
    }
    el.parentNode.className += ' ui raised segment';
    // wrap everything except the h1 in a section into <div class="content">
    d = document.createElement('div');
    d.className = 'content';
    while (el.nextSibling) {
      d.append(el.nextSibling);
    }
    el.parentNode.append(d);
  }

  // turn h2 sections in <div class="member-cards"> into semantic ui's cards
  els = document.querySelectorAll('.member-cards');
  for (i = 0; i < els.length; i++) {
    el = els[i].querySelectorAll('h2');
    for (j = 0; j < el.length; j++) {
      s = el[j];
      d = document.createElement('div');
      d.className = 'ui card';
      d.innerHTML = '<div class="image"></div><div class="content">' +
        '<div class="header"></div><div class="meta"></div>' +
        '<div class="description"></div></div>'
      d.firstChild.innerHTML = s.nextElementSibling.querySelector('img').outerHTML;
      s.parentNode.removeChild(s.nextElementSibling);
      d.lastChild.children[0].innerHTML = s.innerHTML;
      d.lastChild.children[1].innerHTML = s.nextElementSibling.innerHTML;
      s.parentNode.removeChild(s.nextElementSibling);
      d.lastChild.children[2].innerHTML = s.nextElementSibling.innerHTML;
      s.parentNode.removeChild(s.nextElementSibling);
      s.parentNode.replaceChild(d, s);
    }
  }

  // remove the section divs for h2 - h5
  els = document.querySelectorAll('h2, h3, h4, h5');
  for (i = 0; i < els.length; i++) {
    el = els[i].parentNode;
    if (el && el.tagName === 'DIV' && /section level/.test(el.className)) {
      el.outerHTML = el.innerHTML;
    }
  }

  // replace <p><span class="foo">bar</span></p> with <div class="foo">bar</div>
  els = document.getElementsByTagName('p');
  for (i = 0; i < els.length;) {
    el = els[i]; i++;
    if (el.childElementCount !== 1) continue;
    s = el.firstElementChild;
    if (s.tagName !== 'SPAN' || s.outerHTML !== el.innerHTML) continue;
    d = document.createElement('div');
    d.innerHTML = s.innerHTML;
    d.className = s.className;
    el.parentNode.replaceChild(d, el);
    i--;
  }

  // break words in <a> if the text and uri are identical
  els = document.getElementsByTagName('a');
  for (i = 0; i < els.length; i++) {
    el = els[i];
    if (el.getAttribute('href') === el.innerText) el.style.wordBreak = 'break-all';
  }
})();
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = `https://mathjax.rstudio.com/latest/MathJax.js` + "?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>


  </body>
</html>
